pyuvdata/uvbeam/uvbeam.py

```diff
@@ -883,9 +1075,14 @@ def peak_normalize(self):
                 "normalization is not yet implemented"
             )
         for i in range(self.Nfreqs):
-            max_val = abs(self.data_array[:, :, :, i, :]).max()
-            self.data_array[:, :, :, i, :] /= max_val
-            self.bandpass_array[:, i] *= max_val
+            if self.future_array_shapes:
+                max_val = abs(self.data_array[:, :, i, :]).max()
+                self.data_array[:, :, i, :] /= max_val
+                self.bandpass_array[i] *= max_val
+            else:
+                max_val = abs(self.data_array[:, :, :, i, :]).max()
+                self.data_array[:, :, :, i, :] /= max_val
+                self.bandpass_array[:, i] *= max_val
```

Comment:
I think you could use `...` here again as well.

Reply:
I can't here either because of the variation in dimensionality between healpix and regularly gridded beams.